x-pack/plugins/ingest_pipelines/public/application/components/pipeline_editor/components/processor_form/processors/community_id.tsx

```diff
@@ -0,0 +1,298 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { FunctionComponent } from 'react';
+import { i18n } from '@kbn/i18n';
+import { FormattedMessage } from '@kbn/i18n/react';
+import { EuiSpacer, EuiCode, EuiFlexGroup, EuiFlexItem } from '@elastic/eui';
+
+import { FieldsConfig, from } from './shared';
+import { TargetField } from './common_fields/target_field';
+import { IgnoreMissingField } from './common_fields/ignore_missing_field';
+import {
+  Field,
+  UseField,
+  useFormData,
+  FIELD_TYPES,
+  NumericField,
+  SerializerFunc,
+  fieldFormatters,
+  fieldValidators,
+} from '../../../../../../shared_imports';
+
+const SEED_MIN_VALUE = 0;
+const SEED_MAX_VALUE = 65535;
+
+const seedValidator = {
+  max: fieldValidators.numberSmallerThanField({
+    than: SEED_MAX_VALUE,
+    allowEquality: true,
+    message: i18n.translate('xpack.ingestPipelines.pipelineEditor.communityId.seedMaxNumberError', {
+      defaultMessage: `This number must be equal or less than {maxValue}.`,
+      values: { maxValue: SEED_MAX_VALUE },
+    }),
+  }),
+  min: fieldValidators.numberGreaterThanField({
+    than: SEED_MIN_VALUE,
+    allowEquality: true,
+    message: i18n.translate('xpack.ingestPipelines.pipelineEditor.communityId.seedMinNumberError', {
+      defaultMessage: `This number must be equal or greater than {minValue}.`,
+      values: { minValue: SEED_MIN_VALUE },
+    }),
+  }),
+};
+
+const fieldsConfig: FieldsConfig = {
+  source_ip: {
+    type: FIELD_TYPES.TEXT,
+    serializer: from.emptyStringToUndefined,
+    label: i18n.translate('xpack.ingestPipelines.pipelineEditor.communityId.sourceIpLabel', {
+      defaultMessage: 'Source IP (optional)',
+    }),
+    helpText: (
+      <FormattedMessage
+        id="xpack.ingestPipelines.pipelineEditor.communityId.sourceIpHelpText"
+        defaultMessage="Field containing the source IP address."
+      />
+    ),
+  },
```

Comment:
May not be possible, but it'd be great to pre-fill these with the default values. I imagine most users will be using the defaults.

If not, we should point out the default field in the help text like we do for target field.

Reply:
Good suggestion! I decided to add the default value to each field help text instead of pre-filling, in order to keep it consistent with the other processors.